airflow/api_fastapi/core_api/datamodels/task_instances.py

```diff
@@ -150,3 +151,44 @@ class TaskInstanceHistoryCollectionResponse(BaseModel):
 
     task_instances: list[TaskInstanceHistoryResponse]
     total_entries: int
+
+
+class SetTaskInstancesStateBody(BaseModel):
+    """Request body for Set Task Instances State endpoint."""
+
+    dry_run: bool = True
+    task_id: str
+    dag_run_id: str
+    include_upstream: bool
+    include_downstream: bool
+    include_future: bool
+    include_past: bool
+    new_state: str
```

Comment:
It would be great to include editing the Task Instance note at the same time like we do for changing a dag run.

Reply:
Right, setting note is covered in https://github.com/apache/airflow/pull/44223